File: Third year/First Semester/Programare pentru dispozitive mobile/Labs/native/app/src/main/java/com/fitlog/app/data/LoggedFood.kt
package com.fitlog.app.data

data class LoggedFood(
    val id: String, // Unique ID for this specific log entry
    val food: FoodItem,
    var grams: Int
) {
    val kcalTotal get() = (food.kcal / 100.0 * grams).toInt()
    val proteinTotal get() = food.protein / 100.0 * grams
    val carbsTotal get() = food.carbs / 100.0 * grams
    val fatTotal get() = food.fat / 100.0 * grams
}

File: Third year/First Semester/Programare pentru dispozitive mobile/Labs/non-native/lib/models/logged_food.dart
import 'food_item.dart';

class LoggedFood {
  const LoggedFood({
    this.id,
    required this.foodId,
    required this.food,
    required this.grams,
  });

  final int? id;
  final int foodId;
  final FoodItem food;
  final int grams;

  int get kcalTotal => (food.kcal / 100 * grams).round();
  double get proteinTotal => food.protein / 100 * grams;
  double get carbsTotal => food.carbs / 100 * grams;
  double get fatTotal => food.fat / 100 * grams;

  LoggedFood copyWith({
    int? id,
    int? foodId,
    FoodItem? food,
    int? grams,
  }) {
    return LoggedFood(
      id: id ?? this.id,
      foodId: foodId ?? this.foodId,
      food: food ?? this.food,
      grams: grams ?? this.grams,
    );
  }
}

